packages/runtime/src/main/kotlin/elide/tool/cli/cmd/repl/ToolShellCommand.kt

```diff
@@ -321,8 +308,24 @@ import elide.tool.project.ProjectManager
       }
     }
 
+    private val tsAliases = setOf("ts", "typescript")
+    private val pyAliases = setOf("py", "python")
+    private val rbAliases = setOf("rb", "ruby")
+    private val jvmAliases = setOf("jvm", "java")
+    private val ktAliases = setOf("kt", "kotlin")
+
     // Resolve the specified language.
-    internal fun resolve(project: ProjectInfo? = null): EnumSet<GuestLanguage> = langs
+    internal fun resolve(project: ProjectInfo? = null, alias: String? = null): EnumSet<GuestLanguage> {
+      return EnumSet.noneOf(GuestLanguage::class.java).apply {
+        add(JS)
+        add(WASM)
+        if (ENABLE_TYPESCRIPT && (typescript || (alias != null && tsAliases.contains(alias)))) add(TYPESCRIPT)
+        if (ENABLE_PYTHON && (python || (alias != null && pyAliases.contains(alias)))) add(PYTHON)
+        if (ENABLE_RUBY && (ruby || (alias != null && rbAliases.contains(alias)))) add(RUBY)
+        if (ENABLE_JVM && (jvm || kotlin || (alias != null && jvmAliases.contains(alias)))) add(JVM)
+        if (ENABLE_JVM && (kotlin || (alias != null && ktAliases.contains(alias)))) add(KOTLIN)
```

Comment:
Languages are now "activated" (and then initialized) iff:

(1) The language is supported
(2) The language is active by default (`JS`, `WASM`), or is requested by the user (all others)